packages/node-sdk/src/client.ts

```diff
@@ -367,6 +367,67 @@ export class BucketClient {
     });
   }
 
+  /**
+   * Updates the context in Bucket (if needed).
+   * This method should be used before requesting feature flags or binding a client.
+   *
+   * @param context - The context to update.
+   */
+  private async syncContext(context: Context) {
+    ok(isObject(context), "context must be an object");
+    ok(
+      typeof context.enableTracking === "boolean" ||
+        context.enableTracking === undefined,
+      "enableTracking must be a boolean",
+    );
+
+    if (context.enableTracking === false) {
+      this._config.logger?.debug(
+        "feature.syncContext(): tracking disabled from context, not updating user/company",
+      );
+
+      return;
+    }
+
+    const promises: Promise<void>[] = [];
+    if (
+      context.company &&
+      this._config.rateLimiter.isAllowed(
+        JSON.stringify(context.company, null, 0),
+      )
+    ) {
+      const { id: _, ...attributes } = context.company;
+      promises.push(
+        this.updateCompany(context.company.id, {
```

Comment:
lets check `isAllowed` inside `updateCompany`, `updateUser` etc. instead. Same for `enableTracking`. Should simplify things